hooks/useVotingGuide.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  Exact,
+  useElectionVotingGuideByUserIdQuery,
+  useUpsertVotingGuideMutation,
+  VotingGuideResult,
+} from "generated";
+import { createContext, ReactNode, useContext } from "react";
+import { useQueryClient } from "react-query";
+
+type VotingGuideQueryContext = {
+  data: VotingGuideResult;
+  queryKey: (string | Exact<{ electionId: string; userId: string }>)[];
+};
+
+const VotingGuideContext = createContext<VotingGuideQueryContext>(
+  {} as VotingGuideQueryContext
+);
+
+export function VotingGuideProvider({
+  electionId,
+  userId,
+  children,
+}: {
+  electionId: string;
+  userId: string;
+  children: ReactNode;
+}) {
+  const { data, isLoading, error } = useElectionVotingGuideByUserIdQuery({
+    electionId,
+    userId,
+  });
+
+  const queryKey = useElectionVotingGuideByUserIdQuery.getKey({
+    electionId,
+    userId,
+  });
+
+  const queryClient = useQueryClient();
+
+  const createVotingGuide = useUpsertVotingGuideMutation({
+    onSuccess: () => queryClient.invalidateQueries(queryKey),
+  });
+
+  if (!data?.electionVotingGuideByUserId)
+    createVotingGuide.mutate({ electionId: electionId as string });
```

Comment:
I see what you mean here with the two calls. I don't really like the idea of creating an object during the get call on line 28 so unless we can create the voting guide object when they first add a candidate to it for an election, this is what we live with.

Reply:
Yea i think running the `createVotingGuide` mutation as a setup in the VotingGuideContext, followed by the `upsertVotingGuideCandidate` methods when the user adds an endorsement is _a decent start._  I was however thinking a user may not want to create a guide at all and perhaps we want to allow them to manually instantiate one or call the `createVotingGuide` the first time they endorse/note a candidate? Should be doable